src/services/TestResultsService.ts

```diff
@@ -431,9 +403,9 @@ export class TestResultsService {
             testType,
             vehicleType: enums.VEHICLE_TYPE[payload.vehicleType.toUpperCase() as keyof typeof enums.VEHICLE_TYPE],
             recentExpiry,
-            regnDate: this.getRegnDateByVehicleType(payload),
+            regnDate: this.getRegistrationOrFirstUseDate(payload),
```

Comment:
In this case I would change the field name also to indicate that it might contain first use or registration date